tests/integration/src/test/java/org/mule/test/core/context/notification/processors/ModuleComponentPathTestCase.java

```diff
@@ -0,0 +1,440 @@
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.runtime.api.component.TypedComponentIdentifier;
+import org.mule.runtime.api.dsl.DslResolvingContext;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.DefaultMuleContext;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.ConfigurationBuilder;
+import org.mule.runtime.core.config.builders.AbstractConfigurationBuilder;
+import org.mule.runtime.core.context.notification.MessageProcessorNotification;
+import org.mule.runtime.dsl.api.component.config.DefaultComponentLocation;
+import org.mule.runtime.extension.internal.loader.XmlExtensionModelLoader;
+import org.mule.runtime.module.extension.internal.manager.DefaultExtensionManager;
+import ru.yandex.qatools.allure.annotations.Description;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+@Description("Test suite to ensure XML connectors do properly generate the paths")
+public class ModuleComponentPathTestCase extends MuleArtifactFunctionalTestCase {
+
+  private static final String COLON_SEPARATOR = ":";
+  private static final String MODULE_SIMPLE_XML = "module-simple.xml";
+  private static final String MODULE_SIMPLE_PROXY_XML = "module-simple-proxy.xml";
+  private static final String FLOWS_USING_MODULE_SIMPLE_XML = "flows-using-modules.xml";
+  private static final String BASE_PATH_XML_MODULES = "org/mule/test/integration/notifications/modules/";
+
+  @Override
+  protected String getConfigFile() {
+    return BASE_PATH_XML_MODULES + FLOWS_USING_MODULE_SIMPLE_XML;
+  }
+
+  private static final Optional<String> CONFIG_FILE_NAME = of(FLOWS_USING_MODULE_SIMPLE_XML);
+  private static final Optional<String> MODULE_SIMPLE_FILE_NAME = of(MODULE_SIMPLE_XML);
+  private static final Optional<String> MODULE_SIMPLE_PROXY_FILE_NAME = of(MODULE_SIMPLE_PROXY_XML);
+  private static final Optional<TypedComponentIdentifier> FLOW_TYPED_COMPONENT_IDENTIFIER =
+      of(builder().withIdentifier(FLOW_IDENTIFIER).withType(FLOW).build());
+
+  private static final DefaultComponentLocation FLOW_WITH_SINGLE_MP_LOCATION =
+      new DefaultComponentLocation(of("flowWithSingleMp"),
+                                   asList(new DefaultComponentLocation.DefaultLocationPart("flowWithSingleMp",
+                                                                                           FLOW_TYPED_COMPONENT_IDENTIFIER,
+                                                                                           CONFIG_FILE_NAME,
+                                                                                           of(15))));
```

Comment:
they are all the line numbers of the operation or flows, I was mimicking `org.mule.test.core.context.notification.processors.ComponentPathTestCase` to make the tests look alike when fixing issues.

Which one is right?